rllib/core/learner/learner.py

```diff
@@ -266,69 +266,85 @@ class Learner:
-        results = learner.update(batch)
+            # Create a learner instance that will train the module
+            learner = TorchLearner(module_spec=module_spec)
+
+            # Note: the learner should be built before it can be used.
+            learner.build()
 
-        # remove a module
-        learner.remove_module("new_player")
+            # Take one gradient update on the module and report the results
+            # results = learner.update(...)
 
-        # will train previous modules only.
-        results = learner.update(batch)
+            # Add a new module, perhaps for league based training
+            learner.add_module(
+                module_id="new_player",
+                module_spec=SingleAgentRLModuleSpec(
+                    module_class=PPOTorchRLModule,
+                    observation_space=env.observation_space,
+                    action_space=env.action_space,
+                    model_config_dict = {"hidden": [128, 128]},
+                    catalog_class = PPOCatalog,
+                )
+            )
 
-        # get the state of the learner
-        state = learner.get_state()
+            # Take another gradient update with both previous and new modules.
+            # results = learner.update(...)
 
-        # set the state of the learner
-        learner.set_state(state)
+            # Remove a module
+            learner.remove_module("new_player")
 
-        # get the weights of the underly multi-agent RLModule
-        weights = learner.get_module_state()
+            # Will train previous modules only.
+            # results = learner.update(...)
 
-        # set the weights of the underly multi-agent RLModule
-        learner.set_module_state(weights)
+            # Get the state of the learner
+            state = learner.get_state()
 
+            # Set the state of the learner
+            learner.set_state(state)
 
-    Extension pattern:
+            # Get the weights of the underly multi-agent RLModule
+            weights = learner.get_module_state()
 
-        .. code-block:: python
+            # Set the weights of the underly multi-agent RLModule
+            learner.set_module_state(weights)
 
-        class MyLearner(TorchLearner):
 
-            def compute_loss(self, fwd_out, batch):
-                # compute the loss based on batch and output of the forward pass
-                # to access the learner hyper-parameters use `self._hps`
+    Extension pattern:
 
-                return {ALL_MODULES: loss}
+        .. doctest::
+
+            >>> from ray.rllib.core.learner.torch.torch_learner import TorchLearner
+
+            >>> class MyLearner(TorchLearner):
+            ...
+            ...    def compute_loss(self, fwd_out, batch):
+            ...        # compute the loss based on batch and output of the forward pass
+            ...        # to access the learner hyper-parameters use `self._hps`
+            ...        return {ALL_MODULES: loss}
```

Comment:
Since this is multiple lines and indented, this might be more readable as `testcode`